python/main.py

```diff
@@ -19,15 +21,66 @@
 )
 
 
+# GET endpoint
 @app.get("/")
 def root():
     return {"message": "Hello, world!"}
 
 
+# function to save given item into json file
+def save_json(new_item, filename='items.json'):
+    if os.path.exists(filename):
+        with open(filename, 'r+') as file:
+            file_data = json.load(file)
+            file_data["items"].append(new_item)
+            file.seek(0)
+            json.dump(file_data, file, indent=4)
+    else:
+        with open(filename, 'w') as file:
+            json.dump({"items": [new_item]}, file, indent=4)
+
+
+# function to save given image into images file
+def save_image(image_bytes, image_name):
+    path = os.path.join("images", image_name)
+    with open(path, "wb") as image:
+        image.write(image_bytes)
+
+
+# POST endpoint for /items
 @app.post("/items")
-def add_item(name: str = Form(...)):
-    logger.info(f"Receive item: {name}")
-    return {"message": f"item received: {name}"}
+def add_item(name: str = Form(), category: str = Form(), image: UploadFile = File()):
+    # (3-4) Save image to directory
+    image_bytes = image.file.read()
+    image_name = f"{hashlib.sha256(image_bytes).hexdigest()}.jpg"
+    save_image(image_bytes, image_name)
+
+    # (3-2) Save item to json
+    item = {"name": name, "category": category, "image_name": image_name}
+    save_json(new_item=item)
+
+    logger.info(f"Receive item: {name}, {category}, {image_name}")
+    return {"message": f"item received: {item}"}
+
+
+# (3-3) GET endpoint for /items
+@app.get("/items")
+def get_items_list(filename='items.json'):
+    with open(filename, 'r') as file:
+        file_data = json.load(file)
+    return {"message": file_data}
+
+
+# (3-5) GET endpoint for /items/{item_id}
+@app.get("/items/{item_id}")
+def get_item_id(item_id: int):
+    with open('items.json', 'r') as file:
+        file_data = json.load(file)
+        items_list = file_data["items"]
+    if 0 < item_id <= len(items_list):
```

Comment:
Could you test this API? I think this condition is opposite from your intention.